sound/soc/sof/intel/telemetry.h

```diff
@@ -0,0 +1,223 @@
+	uint8_t preventedby_fastmode_task;
+	uint8_t prevent_autonomus_reset_counter;
+    uint8_t rsvd[2];
+};
+
+struct transition_info_s {
+	uint32_t entry_cnt;
+	uint32_t exit_cnt;
+	uint32_t last_entry_systick;
+};
+
+struct vad_kpd_stats_s {
+    uint32_t vad_cnt;
+    uint32_t aad_cnt;
+    uint32_t nld_cnt;
+    uint32_t kpd_cnt;
+};
+
+struct ulp_telemetry_s {
+	struct ulp_prevention_info_s prevention_info;
+	struct transition_info_s     transition_info;
+	struct vad_kpd_stats_s       vad_kpd_stats;
+};
+
+struct thread_entry_point_info_s {
+	uint32_t call_stack;
+	uint32_t start_systick;
+};
+
+struct task_info_s {
+	uint32_t task_id;
+	uint8_t task_state;
+	uint8_t task_type;
+	uint8_t task_subtype;
+	uint8_t core_id;
+	struct thread_entry_point_info_s blockade_entry_point;
+	struct thread_entry_point_info_s cs_entry_point;
+};
+
+struct CoreExceptionRecord {
+	uint32_t version;
+	uint32_t stackdump_completion;
+	uint64_t timestamp;
+	uint32_t rec_state;
+	uint32_t exec_ctx;
+	uint32_t epc_1;
+	uint32_t eps_2;
+	uint32_t epc_2;
+	uint32_t depc;
+	uint32_t debugcause;
+	uint32_t exccause;
+	uint32_t excvaddr;
+	uint32_t excsave;
+	uint32_t interrupt;
+	uint32_t ar[SOF_IPC4_FW_AREGS_COUNT];
+	uint32_t windowbase;
+	uint32_t windowstart;
+	uint32_t mem_epc[4];
+	uint32_t stack_base_addr;
+};
+
+#define SOF_TELEMETRY_SEPARATOR_1	0x0000C0DE
+#define SOF_TELEMETRY_SEPARATOR_2	0X0DEC0DE2
+#define SOF_TELEMETRY_SEPARATOR_3	0X0DEC0DE3
+#define SOF_TELEMETRY_SEPARATOR_4	0X0DEC0DE4
+#define SOF_TELEMETRY_SEPARATOR_5	0X0DEC0DE5
+#define SOF_TELEMETRY_SEPARATOR_6	0X0DEC0DE6
+#define SOF_TELEMETRY_SEPARATOR_7	0X0DEC0DE7
+#define SOF_TELEMETRY_SEPARATOR_8	0X0DEC0DE8
+#define SOF_TELEMETRY_SEPARATOR_9	0X0DEC0DE9
+#define SOF_TELEMETRY_SEPARATOR_10	0X0DEC0DEA
+#define SOF_TELEMETRY_SEPARATOR_11	0X0DEC0DEB
+#define SOF_TELEMETRY_SEPARATOR_EXP	0x00E0DE0D
+
+struct TelemetryWndData {
+	uint32_t            separator_1;
+	struct system_tick_info_s  system_tick_info[SOF_IPC4_FW_REPORTED_MAX_CORES_COUNT];
+	uint32_t            separator_2;
+	struct deadlock_info_s     deadlock_info[SOF_IPC4_FW_REPORTED_MAX_CORES_COUNT];
+	uint32_t            separator_3;
+	struct assert_info_s       assert_info;
+	uint32_t            separator_4;
+	struct xxxruns_info_s      xxxruns_info;
+	uint32_t            separator_5;
+	struct performance_info_s  performance_info;
+	uint32_t            separator_6;
+	struct mem_pools_info_s    mem_pools_info;
+	uint32_t            separator_7;
+	struct timeout_info_s      timeout_info;
+	uint32_t            separator_8;
+	struct ulp_telemetry_s     ulp_telemetry;
+	uint32_t            separator_9;
+	struct transition_info_s   evad_transition_info;
+	uint32_t            separator_10;
+	struct task_info_s         task_info[SOF_IPC4_FW_MAX_REPORTED_TASKS];
+	uint32_t            separator_11;
+    struct transition_info_s	d0i3_info;
+    uint32_t	__pad_for_exception_record;
+    uint32_t	separator_exception;
```

Comment:
indentation is off?

Reply:
yes, I will check my code editor setting.